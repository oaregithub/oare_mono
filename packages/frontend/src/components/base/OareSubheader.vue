<template>
  <h3 class="title font-weight-regular mb-4">
    <slot></slot>
    <v-divider />
  </h3>
</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api';

export default defineComponent({
  name: 'OareSubheader',
});
</script>

<style></style>
