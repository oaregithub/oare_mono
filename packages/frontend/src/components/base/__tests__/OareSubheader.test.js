import Vuetify from "vuetify";
import VueCompositionApi from "@vue/composition-api";
import { mount, createLocalVue } from "@vue/test-utils";
import OareSubheader from "@/components/base/OareSubheader";

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe("OareSubheader test", () => {
  const createWrapper = (label) =>
    mount(OareSubheader, {
      vuetify,
      localVue,
      slots: {
        default: label,
      },
    });

  it("matches snapshot", () => {
    expect(createWrapper("Test Subheader")).toMatchSnapshot();
  });

  it("displays passed in slot", () => {
    const text = "Test Subheader";
    expect(createWrapper(text).text()).toBe(text);
  });
});
