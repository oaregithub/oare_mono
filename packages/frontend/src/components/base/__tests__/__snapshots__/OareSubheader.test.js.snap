// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OareSubheader test matches snapshot 1`] = `
<h3 class="title font-weight-regular mb-4">Test Subheader
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
</h3>
`;
